prov/util/src/util_wait.c

```diff
@@ -424,3 +428,214 @@ int ofi_wait_fd_open(struct fid_fabric *fabric_fid, struct fi_wait_attr *attr,
 	free(wait);
 	return ret;
 }
+
+static void util_wait_yield_signal(struct util_wait *util_wait)
+{
+	(container_of(util_wait, struct util_wait_yield,
+			util_wait))->signal = 1;
```

Comment:
There's no protection used around setting signal.  In yield_try, we set it to 0 always.  Is it possible to miss a signal, with this setting it to 1 immediately before yield_try sets it back to 0?

Reply:
At first I was thinking it would be protected since we hold the wait lock while trying all the fids but yeah I think you're right. I think we're going to have to have 2 lock, one for the wait object itself and one just for the signal to avoid missing it.